Tests/base/KVC/nsnull.m

```diff
@@ -0,0 +1,31 @@
+#import "ObjectTesting.h"
+#import <Foundation/NSNull.h>
+#import <Foundation/NSAutoreleasePool.h>
+#import <Foundation/NSKeyValueCoding.h>
+
+int main(void) {
+    NSAutoreleasePool *arp = [NSAutoreleasePool new];
+
+    NSNull *nullObject = [NSNull null];
+
+    // Accessing an undefined key
+    id result = [nullObject valueForKey:@"undefinedKey"];
+    PASS(result == nullObject, "NSNull returns itself for undefined keys.");
+
+    // Attempting to set a value for an undefined key
+    PASS_EXCEPTION([nullObject setValue:@"value" forKey:@"undefinedKey"],
+	          NSUndefinedKeyException,
+              "Setting an undefined key on NSNull should not crash.");
+
+    // Accessing an undefined key path
+    id result = [nullObject valueForKeyPath:@"some.path"];
```

Comment:
Here gcc complains that you are redefining `result` and I think this complain is correct.

Reply:
You are right. I tested something and forgot to change it back :)